test/jdk/com/sun/jndi/ldap/LdapSSLHandshakeFailureTest.java

```diff
@@ -0,0 +1,242 @@
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import javax.net.SocketFactory;
+import javax.net.ssl.SSLServerSocketFactory;
+import javax.naming.ldap.InitialLdapContext;
+import javax.naming.ldap.LdapContext;
+import javax.naming.Context;
+import java.net.SocketException;
+import java.security.KeyStore;
+import java.util.Hashtable;
+
+import jdk.test.lib.net.URIBuilder;
+
+/*
+ * @test
+ * @bug 8314063
+ * @library /test/lib
+ * @summary For LDAPs connection, if the value of com.sun.jndi.ldap.connect.timeout is
+ * set too small or not an optimal value for the system, after the socket is created and
+ * connected to the server, but the handshake between the client and server fails due to
+ * socket time out, the opened socket is not closed properly. In this test case, the server
+ * is forced to sleep ten seconds and connection time out for client is one second. This
+ * will allow the socket opened and connected, and give the chance for the handshake to be
+ * timed out. Before this fix, the socket is kept opened. Right now the exception will be
+ * caught and the socket will be closed.
+ *
+ * @run main/othervm LdapSSLHandshakeFailureTest LdapSSLHandshakeFailureTest$CustomSocketFactory true
+ * @run main/othervm LdapSSLHandshakeFailureTest -1000 true
+ * @run main/othervm LdapSSLHandshakeFailureTest -1000 false
+ * @run main/othervm LdapSSLHandshakeFailureTest 2000 false
+ * @run main/othervm LdapSSLHandshakeFailureTest 0 true
+ * @run main/othervm LdapSSLHandshakeFailureTest 0 false
+ * @run main/othervm LdapSSLHandshakeFailureTest true
+ * @run main/othervm LdapSSLHandshakeFailureTest false
+ */
+
+public class LdapSSLHandshakeFailureTest {
+    private static String SOCKET_CLOSED_MSG = "The socket has been closed.";
+    private static String SOCKET_NOT_CLOSED_MSG = "The socket was not closed.";
+
+    public static void main(String args[]) throws Exception {
+
+        // Set the keystores
+        setKeyStore();
+        // start the test server first.
+        boolean serverSlowDown = false;
+        if(args.length ==2 ) {
+            serverSlowDown = Boolean.valueOf(args[1]);
+        } else {
+            if(args.length ==1 ) {
+                serverSlowDown = Boolean.valueOf(args[0]);
+            }
+        }
+        TestServer server = new TestServer( serverSlowDown );
+        server.start();
+        Hashtable<String, Object> env = new Hashtable<>();
+        env.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.ldap.LdapCtxFactory");
+        env.put(Context.PROVIDER_URL, URIBuilder.newBuilder()
+                .scheme("ldaps")
+                .loopback()
+                .port(server.getPortNumber())
+                .buildUnchecked().toString());
+        if (args.length == 2 &&
+                args[0].contains("LdapSSLHandshakeFailureTest")) {
```

Comment:
This `arg[0]` check is done twice in the test - it can be save to a boolean variable:
```
boolean hasCustomSocketFactory = args[0]
        .equals("LdapSSLHandshakeFailureTest$CustomSocketFactory");
```
and then used to the checks